### PR TITLE
Hide branch name from title bar when there are no commits ahead of master.

### PR DESCRIPTION
**Pull request by jordan-woyak**
Package maintainer build systems are building from alternate branches for whatever reason.
e.g. The build in the official Arch Linux repository *is* 5.0-20347 based on the commit ID, but shows `[makepkg]` in the title bar, which is ugly. This PR should fix this problem:
image

`SCM_IS_MASTER` is replaced with `SCM_COMMITS_AHEAD_MASTER` to detect if the current branch is equivalent to some commit in master.
The branch name is now hidden from the title bar unless there are actually commits ahead of master.
image

The number of commits ahead of master is also shown in the About dialog.
image

edit: If the master branch does not exist then the `[branch-name]` will be shown in the title bar and the "commit(s) ahead" About-message will **not** be shown.

### CODE REVIEW

File: Source/Core/DolphinQt/AboutDialog.cpp [hunk 1]
```diff
@@ -17,6 +17,17 @@ AboutDialog::AboutDialog(QWidget* parent) : QDialog(parent)
...
 
+  QString branch_str = QString::fromStdString(Common::GetScmBranchStr());
+  const int commits_ahead = Common::GetScmCommitsAheadMaster();
+  if (commits_ahead > 0)
+  {
+    branch_str +=
+        QStringLiteral(" (") +
+        // i18n: A positive number of version control commits made compared to some named branch
+        tr("%1 commit(s) ahead of %2").arg(commits_ahead).arg(QStringLiteral("master")) +
+        QStringLiteral(")");
```
Comment (JosJuice) [inline Source/Core/DolphinQt/AboutDialog.cpp:28]: String concatenation isn't translation friendly. Instead, I would recommend "%1 (%2)".arg(...).arg(...)

**Comment by endrift** [comment]
This raises the question: how do you make sure their master is the same master? This broke on the FreeBSD builder because that one has never had its own local version of master, only origin/master. Who's to say similar things won't occur downstream? You'd also need to make sure that Central keeps local master in sync with origin master.

**Comment by jordan-woyak** [comment]
> This raises the question: how do you make sure their master is the same master? This broke on the FreeBSD builder because that one has never had its own local version of master, only origin/master. Who's to say similar things won't occur downstream? You'd also need to make sure that Central keeps local master in sync with origin master.

It's something to think about, but this PR hasn't changed that. It adds some more information when available. It's always going to be possible to fudge that information in various ways. A person could commit directly to local master to keep the title bar from changing and of course nothing is stopping a person from changing the version text source code to display literally anything that they want. 

The end result remains unchanged in this PR for FreeBSD, the branch name will be displayed and the About dialog contains no additional information.

**Review by Tilka — APPROVED** [review]
Untested but looks good. If master does not exist, git will print an error to stderr and stdout will be empty.

**Pull request merged**